### ISSUE TITLE
Implement scikit-learn like API

### ISSUE THREAD

**Issue by cbrnr** [scot-dev/scot#187]
It would be nice to have an API similar to that of `scikit-learn`, e.g. to perform MVARICA:

```{Python}
model = scot.MVARICA()
model.fit(data)
```

We'd have to discuss the details though, such as where do we specify certain parameters (e.g. model order, dimensionality - in `__init__` or `fit`). @mbillingr I would be happy if you had some comments on that.

**Comment by mbillingr** [comment]
I like the idea. In accordance with scikit-learn, I'd say most configuration parameters should go into `__init__`.

We should think about the remaing API too (e.g. what to do with the model after fitting).
- Should there be a `.transform()` method that allows MVARICA to be used in place of an ICA? 
- How do we expose the fitted components (PCA, ICA, VAR) for further use?
- How will it fit in current workflows (online application, connectivity analysis, time-/frequency analysis)?

**Comment by cbrnr** [comment]
I agree with putting most parameters into `__init__`.

I think we definitely need a `transform` method. If this works like in `scikit-learn`, the `transform` method should return the transformed signals and thereby exposing the fitted components. Is this what you meant with your second question?

Regarding the workflows, I think we should develop this new API in parallel to the existing one(s) so that we don't break existing workflows. I'm not particularly concerned with online application, since this is not an explicit goal of SCoT. We should try to fit in as nicely as possible into the existing ecosystem, which is mostly `NumPy`, `SciPy`, and `MNE` in our case. Do we need to get more concrete before starting to work on this new API?

**Comment by mbillingr** [comment]
`transform` returns the components, agreed. 
How do we expose the VAR model for connectivity analysis? I'd suggest a `var` property.

Anything else we need to consider?
